src/libraries/System.Diagnostics.Process/tests/ProcessStreamReadTests.cs

```diff
@@ -56,6 +56,37 @@ public void TestAsyncErrorStream()
             }
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void TestAsyncErrorStream_SynchronizingObject(bool invokeRequired)
+        {
+            StringBuilder sb = new StringBuilder();
+            int invokeCalled = 0;
+
+            Process p = CreateProcessPortable(RemotelyInvokable.ErrorProcessBody);
+            p.SynchronizingObject = new DelegateSynchronizeInvoke()
+            {
+                InvokeRequiredDelegate = () => invokeRequired,
+                InvokeDelegate = (d, args) =>
+                {
+                    invokeCalled++;
+                    return d.DynamicInvoke(args);
+                }
+            };
+            p.StartInfo.RedirectStandardError = true;
+            p.ErrorDataReceived += (s, e) => sb.Append(e.Data);
+            p.Start();
+            p.BeginErrorReadLine();
+
+            Assert.True(p.WaitForExit(WaitInMS));
+            p.WaitForExit(); // This ensures async event handlers are finished processing.
+
+            string expected = RemotelyInvokable.TestConsoleApp + " started error stream" + RemotelyInvokable.TestConsoleApp + " closed error stream";
```

Comment:
nit: it could be const?
```suggestion
            string const expected = RemotelyInvokable.TestConsoleApp + " started error stream" + RemotelyInvokable.TestConsoleApp + " closed error stream";
```

Reply:
Sure